functions/services/firebase/firebase.test.js

```diff
@@ -0,0 +1,309 @@
+            let teamData = {
+                OnFleetID: "<some-fleet-id-2>",
+                zipcode: existingTeam.zipcode
+            };
+
+            const firestore = mocksdk.firestore();
+
+            // Insert existing team
+            firestore.collection("teams").data = {
+                [existingTeam.zipcode]: existingTeam
+            };
+
+            // Assert that the existingTeam is there
+            firestoreTeamsData = firestore.collection("teams").data;
+            expect(firestoreTeamsData).toHaveProperty(
+                existingTeam.zipcode,
+                existingTeam
+            );
+
+            const pr = writeNewTeam(teamData.OnFleetID, teamData.zipcode);
+
+            expect(pr).toResolve();
+
+            // running the waiting tasks (without flush the pr will never resolve)
+            await firestore.flush();
+            await pr;
+
+            // Assert data actually inserted
+            firestoreTeamsData = firestore.collection("teams").data;
+            expect(firestoreTeamsData).toHaveProperty(
+                teamData.zipcode,
+                teamData
+            );
+        });
+
+        describe("Input validation", () => {
+            it("should throw an error when passing no args", () => {
+                expect(writeNewTeam()).toReject();
+            });
+
+            it("should throw an error when passing only 1 argument", () => {
+                expect(writeNewTeam("some-value")).toReject();
+            });
+
+            it("should throw an error when passing undefined argument", () => {
+                expect(writeNewTeam(undefined, undefined)).toReject();
+            });
+
+            it("should throw an error when passing object and undefined arguments", () => {
+                expect(writeNewTeam({}, undefined)).toReject();
+            });
+        });
+    });
+
+    describe("#getTeam", () => {
+        let getTeam;
+        let mocksdk;
+
+        beforeEach(() => {
+            jest.resetModules();
+
+            mocksdk = createMockSdk();
+            jest.mock("firebase", () => mocksdk);
+
+            getTeam = require("./").getTeam;
+        });
+
+        afterEach(() => {
+            getTeam = undefined;
+        });
+
+        it("should get undefined when there is no team", async () => {
+            let firestoreTeamsData;
+
+            let zipcode = "342897";
+
+            const firestore = mocksdk.firestore();
+
+            // Assert no data been before
+            firestoreTeamsData = firestore.collection("teams").data;
+            expect(firestoreTeamsData).toBeNull();
+
+            const pr = getTeam(zipcode);
+
+            expect(pr).toResolve();
+
+            // running the waiting tasks (without flush the pr will never resolve)
+            firestore.flush();
+            const team = await pr;
+
+            expect(team).toBeNull();
+        });
+
+        it("should not get team when Firestore already different team", async () => {
```

Comment:
```suggestion
        it("should not get team when Firestore already has different team", async () => {
```
? Not sure if this was the intention

Reply:
Yeah, thank you